lib/chai/core/assertions.js

```diff
@@ -1069,18 +1069,49 @@ module.exports = function (chai, _) {
     var obj = flag(this, 'object')
       , doLength = flag(this, 'doLength')
       , flagMsg = flag(this, 'message')
-      , ssfi = flag(this, 'ssfi');
+      , msgPrefix = (flagMsg) ? flagMsg + ': ' : ''
+      , ssfi = flag(this, 'ssfi')
+      , objType = _.type(obj).toLowerCase()
+      , nType = _.type(n).toLowerCase();
 
     if (doLength) {
       new Assertion(obj, flagMsg, ssfi, true).to.have.property('length');
     } else {
-      new Assertion(obj, flagMsg, ssfi, true).is.a('number');
+       // Can't use utils/expectTypes because of the error while handling date objects
+      if (!(objType === 'number' || objType === 'date')) {
+        obj = (objType === 'string') 
+          ? '\'' + obj + '\''
+          : obj;
+        throw new AssertionError(
+          msgPrefix + 'expected ' + obj + ' to be a number or a date',
+          undefined,
+          ssfi
+        );
+      }
     }
 
-    if (typeof n !== 'number') {
-      flagMsg = flagMsg ? flagMsg + ': ' : '';
+    if (!(nType === 'number' || nType === 'date')) {
+      var shoudBeType = (objType === 'number' || objType === 'date' )
```

Comment:
@vieiralucas Instead of doing `else-if` I got rid of the nested ternary. If you think that it contradicts a style or obscures code, let me now and I'll change this too.

Restoring the question - if I were to separate this logic elsewhere, what's the best place? Utils?